hack/bin/Sbom.hs

```diff
@@ -0,0 +1,376 @@
+      mkDependencies meta = do
+        let d =
+              object
+                [ "ref" .= meta.outPath,
+                  "dependsOn" .= runIdentity meta.directDeps
+                ]
+        [d]
+      mkComponents :: SBomMeta Identity -> Array
+      mkComponents meta = do
+        let c :: Value
+            c =
+              -- FUTUREWORK(mangoiv): swid? https://www.iso.org/standard/65666.html
+              -- FUTUREWORK(mangoiv): CPE?
+              -- FUTUREWORK(mangoiv): more information in the supplier section
+              object
+                [ "type" .= meta.typ,
+                  "bom-ref" .= String (runIdentity meta.outPath),
+                  "supplier" .= object ["url" .= nubOrd (maybeToList meta.homepage <> catMaybes meta.urls)],
+                  "name" .= String (fromMaybe (st'name $ splitStorePath $ runIdentity meta.outPath) meta.name),
+                  "version" .= meta.version,
+                  "description" .= meta.description,
+                  "scope" .= String "required",
+                  "licenses" .= ((\ln -> object ["license" .= ln]) <$> filter (isJust . (>>= (.id))) meta.licenseSpdxId),
+                  "purl" .= mkPurl meta
+                ]
+        [c]
+      (dependencies, components) = foldMap (mkDependencies &&& mkComponents) bom
+
+  pure $
+    encode @Value $
+      object
+        [ "bomFormat" .= String "CycloneDX",
+          "specVersion" .= String "1.5",
+          "serialNumber" .= String ("urn:uuid:" <> UUID.toText uuid),
+          "version" .= Number (fromIntegral settings.sbom'version),
+          "metadata"
+            .= object
+              [ "timestamp" .= String (T.pack (show curTime)),
+                "component"
+                  .= object
+                    [ "name" .= String settings.sbom'component,
+                      "type" .= String "application"
+                      -- FUTUREWORK(mangoiv): this should be a choice in the settings above
+                    ],
+                -- FUTUREWORK(mangoiv): "manufacture" can also have url
+                "manufacture" .= object ["name" .= String settings.sbom'manufacture],
+                "supplier" .= object ["name" .= String (fromMaybe settings.sbom'manufacture settings.sbom'supplier)],
+                "licenses" .= Array (fromList $ object . (\n -> ["id" .= n]) . String <$> settings.sbom'licenses)
+              ],
+          "components" .= Array components,
+          -- FUTUREWORK(mangoiv): services: allow to tell the program the name of the services like brig, galley, ...
+          "dependencies" .= Array dependencies
+        ]
+
+buildMetaDB :: [Meta] -> MetaDB
+buildMetaDB = foldMap \MkSBomMeta {..} -> [(drvPath, MkSBomMeta {..})]
+
+discoverSBom :: FilePath -> MetaDB -> IO SBom
+discoverSBom outP metaDb = do
+  canonicalOutP <- canonicalizePath =<< getSymbolicLinkTarget outP
+  info <- pathInfo canonicalOutP
+  let go :: (Text, (Text, [Text])) -> IO SBom -> IO SBom
```

Comment:
```suggestion
  let go :: PathInfo -> IO SBom -> IO SBom
```

I'm not sure if we really need `PathInfo` as type synonym. However, as it's there, we can use it consequently :)